api/v1alpha3/azurecluster_webhook.go

```diff
@@ -48,15 +50,51 @@ func (c *AzureCluster) Default() {
 // ValidateCreate implements webhook.Validator so a webhook will be registered for the type
 func (c *AzureCluster) ValidateCreate() error {
 	clusterlog.Info("validate create", "name", c.Name)
-
-	return c.validateCluster()
+	allErrs := c.validateCluster()
+
+	// validation is smart/lazy enough to work with the same object compared to itself
+	// it works well for update to take both objects.
+	if err := validateControlPlaneIP(
+		c.Spec.NetworkSpec.PublicIP,
+		c.Spec.NetworkSpec.PublicIP,
```

Comment:
this seems a bit odd, left a note about it in #777 

I guess it would be hard to pass in a nil object for Create because then it's hard to tell if the user added the field during an update or if it's a new cluster create

Reply:
I could do this?

```go
func (new *PublicIPSpec) validateControlPlaneIP(old *PublicIPSpec, fldPath *field.Path) *field.Error {
	if old == nil && new != nil {
		return field.Invalid(fldPath, new, fmt.Sprintf("setting control plane endpoint after cluster creation is not allowed"))
	}
	if old != nil && new == nil {
		return field.Invalid(fldPath, new, fmt.Sprintf("removing control plane endpoint after cluster creation is not allowed"))
	}
	if old != nil && new != nil && old.Name != new.Name {
		return field.Invalid(fldPath, new, fmt.Sprintf("changing control plane endpoint after cluster creation is not allowed"))
	}
	if new != nil && new.Name == "" {
		return field.Invalid(fldPath, new, fmt.Sprintf("control plane endpoint IP name must be non-empty"))
	}
	return nil
}
```

Reply:
this could be similar to reconciliation -- each type exposes its own validation, we just enumerate them

Reply:
up to you, I think either way is fine.